client.go

```diff
@@ -345,12 +348,15 @@ func (cl *Client) Close() {
 	cl.closed.Set()
 	if cl.dHT != nil {
 		cl.dHT.Close()
+		cl.dHT = nil
 	}
 	if cl.utpSock != nil {
 		cl.utpSock.Close()
+		cl.utpSock = nil
 	}
 	if cl.tcpListener != nil {
 		cl.tcpListener.Close()
+		cl.tcpListener = nil
 	}
 	for _, t := range cl.torrents {
```

Comment:
it does when code start to live its own live. and when it happens it will crash. it self checking nil. for example when you .Drop() torrent it will start live it's own live after receiving half open connection data or tracker announces.